rcl/src/rcl/timer.c

```diff
@@ -233,6 +233,10 @@ rcl_ret_t
 rcl_timer_clock(rcl_timer_t * timer, rcl_clock_t ** clock)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(timer, RCL_RET_INVALID_ARGUMENT);
+  if (timer->impl == NULL) {
+    RCL_SET_ERROR_MSG("timer is invalid");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
```

Comment:
how about the following? so are other places.

```suggestion
  RCL_CHECK_ARGUMENT_FOR_NULL(timer->impl, RCL_RET_INVALID_ARGUMENT);
```